src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Finalization.cs

```diff
@@ -57,5 +57,54 @@ public async Task IncompleteResponseStream_ResponseDropped_CancelsRequestToServe
                 }));
             }
         }
+
+        [Fact]
+        public async Task ClientNotDisposed_ServerReadsFromStream_ServerHangs()
+        {
+            Diagnostics.Debugger.Launch();
+            using (HttpClient client = new HttpClient())
+            {
+                await LoopbackServerFactory.CreateClientAndServerAsync(async url =>
+                {
+                    var stream = await client.GetStreamAsync(url);
+                    stream.Close();
+                    //client.Dispose(); Uncomment this to close the request stream and make the test succeed
+                },
+                server => server.AcceptConnectionAsync(async con =>
+                {
+                    await con.SendResponseAsync(content: "Response");
+
+                    byte[] buf = new byte[1024];
+                    int count = 0;
+                    do
+                    {
+                        count = con.Stream.Read(buf, 0, buf.Length);
+                    } while (count > 0);
+                }));
+            }
+        }
+
+        [Fact]
+        public async Task ClientNotDisposed_ClientSendsSecondRequest_ClientHangs()
+        {
+            using (HttpClient client = new HttpClient())
+            {
+                var requestRead = new AutoResetEvent(false);
+                await LoopbackServerFactory.CreateClientAndServerAsync(async url =>
+                {
+                    var stream = await client.GetStreamAsync(url);
+                    stream.Close();
+                    requestRead.Set();
+                    stream = await client.GetStreamAsync(url);
+                    stream.Close();
+                    requestRead.Set();
+                },
+                server => server.AcceptConnectionAsync(async con =>
+                {
+                    await con.SendResponseAsync(content: "Response");
```

Comment:
Since you are not waiting for request to come. response can go out before client sends request. This should not matter in this case but I saw that ion packet capure and it looks weird.

Reply:
Interesting, on my machine SendResponseAsync is invoked only after the client sends a request (protocol is HTTP 1.1). Moreover, from the TCP point of view, the server connection cannot be created until the client connection request arrives to the server since there is no any client socket to send response to.